sklearn_numba_dpex/common/kernels.py

```diff
@@ -434,6 +464,9 @@ def _make_partial_sum_reduction_2d_axis1_kernel(
     reduction_block_size = 2 * work_group_size
     work_group_shape = (1, work_group_size)
 
+    local_sum_and_set_items_if = _make_sum_and_set_items_if_kernel_func()
+    global_sum_and_set_items_if = _make_sum_and_set_items_if_kernel_func()
```

Comment:
Why do we need two of those and not just one? Is it for this reason?

```suggestion
    # HACK: must define twice to work around the bug highlighted in
    # test_regression_fix
    local_sum_and_set_items_if = _make_sum_and_set_items_if_kernel_func()
    global_sum_and_set_items_if = _make_sum_and_set_items_if_kernel_func()
```